static/app/components/events/interfaces/threadsV2.tsx

```diff
@@ -59,20 +59,46 @@ function Threads({
   hasHierarchicalGrouping,
   groupingCurrentLevel,
 }: Props) {
+  const threads = data.values ?? [];
+
   const [state, setState] = useState<State>(() => {
-    const thread = defined(data.values) ? findBestThread(data.values) : undefined;
+    const thread = !!threads.length ? findBestThread(threads) : undefined;
     return {activeThread: thread};
   });
 
-  const threads = data.values ?? [];
   const stackTraceNotFound = !threads.length;
   const {activeThread} = state;
 
-  const platform = (event.platform ?? 'other') as PlatformType;
   const hasMoreThanOneThread = threads.length > 1;
   const exception = getThreadException(event, activeThread);
   const stackView = activeThread ? getIntendedStackView(activeThread, event) : undefined;
 
+  function getPlatform(): PlatformType {
+    const exceptionValue = exception?.values?.find(
+      value => !!value.stacktrace?.frames?.find(frame => !!frame.platform)
+    );
+
+    if (exceptionValue) {
+      const exceptionFramePlatform = exceptionValue.stacktrace?.frames?.find(
+        frame => !!frame.platform
+      );
```

Comment:
Is there a way to combine these two operations so that you don't have to walk the exception stackframes twice? Or is the platform generally in the first few elements of the frames?

Reply:
yes I can combine the logic :)